### ISSUE TITLE
schema compiler overwrites self-defined resolver input

### ISSUE THREAD

**Issue by avlonder** [aws-amplify/amplify-cli#5715]
**Bug description**

When your index is composed of a primary key and sort key, the graphql compiler sometimes unnecessarily overwrites the inputs defined in the original schema. For example, with the schema:

```gql
type User { ... }

type FriendRequest
  @model(queries: null, mutations: null)
  @key(fields: ["receiverId", "senderId"])
  @key(name: "bySender", fields: ["senderId", "receiverId"]) {
  senderId: ID!
  receiverId: ID!
  sender: User @connection(fields: ["senderId"])
  receiver: User @connection(fields: ["receiverId"])
}

type Mutation {
  createFriendRequest(input: CreateFriendRequestInput!): FriendRequest
  deleteFriendRequest(input: DeleteFriendRequestInput!): FriendRequest
}

input CreateFriendRequestInput {
  receiverId: ID!
}

input DeleteFriendRequestInput {
  receiverId: ID!
}
```
the compiler changes the latter to

```gql
input DeleteFriendRequestInput {
  receiverId: ID!
  senderId: ID!
}
```

However, CreateFriendRequestInput remains untouched.
The senderId is unnecessary because it is based on $ctx.identity.username inside the resolver.

**Amplify CLI Version**
4.13.4

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Node Version: 12.13.1

**Comment by jhockett** [comment]
Hi @avlonder, I noticed you're on a pretty old version of the CLI. At time of writing, the current version is 4.31.1.

I was unable reproduce this on the latest version.

**Comment by edwardfoyle** [comment]
@avlonder I was able to repro this in the latest version of the CLI.

**Comment by avlonder** [comment]
Ok, thanks for the repro! I upgraded the CLI and, indeed, the bug still exists.

**Comment by avlonder** [comment]
Any idea if this is going to be fixed soon?
It's annoying that I have to manually rewrite parts of my schema in the console after every api push.
This is going to lead to errors in production soon or later without a doubt.

**Comment by avlonder** [comment]
Sorry guys, but my particular issue mentioned above still isn't solved with version 4.45.2.
As you can see: apart from a user-defined mutation, I also have a user-defined input type.
Since I'm resolving authentication with $ctx.identity.username inside my custom resolvers, I do not need some of the key fields in my input definitions. As you can see, both my input definitions are missing the senderId and this is on purpose.
But the graphql compiler still overwrites the input types and adds the senderId field, as such yielding errors in my client application. Of course, I can manually remove the senderId in the AppSync console, but sooner or later this is going to result in bugs in production.

**Comment by akshbhu** [comment]
Hi @avlonder 

The changes are behind a featureFlag so that it doesnt affect existing projects. you can set ```skipOverrideMutationInputTypes``` to true . Currently its default value if false for existing projects.

We havent updated the docs site yet for the docs.
Let me know if this works for you?
